### DISCUSSION TITLE
Actividad 04 - GRUPO 7

### DISCUSSION THREAD

**Discussion by bsantillan** [frlp-utn-ingsoft/catedra#157] [Show and tell]
https://github.com/IZelayeta/vetsoft

**Comment by ivanduva** [comment]
Buenas! Les puedo pedir el link a donde lo deployaron?

**Reply by ivanduva** [reply]
Les recuerdo nuevamente pasar el link al deploy

**Reply by bsantillan** [reply]
Buenas profesor, disculpe la demora. 
El link a donde deployamos fue
https://vetsoft-app.onrender.com/

Nuevamente disculpe la demora.

**Comment by RodrigoJacznik** [comment]
Buenas. Les dejo el feedback de la actividad 3 y 4.

Actividad 3:

Se mejoro el nombre de los commits (dejando de lado el formato anterior). 
Vemos que hay 2 branches en donde lo test no funciona en varios commit. Los pr de esos branches son:
- https://github.com/IZelayeta/vetsoft/pull/5
- https://github.com/IZelayeta/vetsoft/pull/8 Este se mergeo con errores que no permiten correr los tests

Estos errores se fixearon directo en main con este commit b1237647972b6787de80b3a75871be6c49c72858. Recuerden que los prs son instancias de revision y no se deberían mergear pr con errores. Tampoco es buena practica fixear directo a main

Tengan cuidado con el copy&paste:
Por ejemplo los test funcionales de este pr: https://github.com/IZelayeta/vetsoft/pull/6. Los nombres de los test funcionales sobre medicine quedaron haciendo referencia a cliente.

Actividad 4:

Me gustó la intención de seguir un formato para el mensaje de los commits. Este formato tiene algunas inconsistencias (a veces referencian archivos y otras carpetas) y puede ser difícil de seguir (que pasa si una feature modifica views.py y models.py?). Les dejo una opción bastante usada como convención para los mensajes: https://www.conventionalcommits.org/en/v1.0.0/#summary

Los PRs los veo bien, las features/bugs tienen sus test y fueron aceptados con el ci en verde (excepto el pr que agrega ci/cd. Ojo, no es una buena practica).
Les dejos algunas criticas sobre los Prs:

- No es común cerrar un PR porque el ci falla. En general se sigue trabajando sobre el mismo
- No es común cerrar un PR sin dejar por lo menos un comentario explicando la causa. Tomen a los PRs como parte de la documentación donde se discute y se iteran soluciones.
- Algunos PRs no son atómicos. Pe: https://github.com/IZelayeta/vetsoft/pull/11 donde se aplica una regla de ruff y se cambia nombres de test.

@bsantillan. Probe la funcionalidad que agregaste (El teléfono debe empezar con 54) y no me muestra el mensaje de error. Por favor arregla este bug y agrega el test correspondiente.

Tenes tiempo hasta el 08/07/2024 para resolverlo

En general creo que hicieron un buen trabajo

**Reply by bsantillan** [reply]
Buenas noches, ya se corrigió el bug que nos remarco anteriormente.
El link nuevamente al repositorio es: https://github.com/IZelayeta/vetsoft.git
Y el link del Deploy es: https://vetsoft-app.onrender.com/
